server/src/routes/posts.js

```diff
@@ -0,0 +1,55 @@
+const express = require('express');
+const router = express.Router();
+
+const { User, Board, Post } = require('../models');
+
+const token = require('../middleware/token');
+
+const _ = require('lodash');
+
+//authenticated routes below this middleware
+router.use(token());
+
+// @route    GET users/posts
+// @desc     Get all posts based on user's interest or filter
+// @access   Private
+router.get('/', async (req, res) => {
+    const {search_filter = '', easy_filters = ''} = req.query;
+    try {
+        const promises = [
+            Post.find({}, (err, post) => {
+                if (err) {
+                    res.status(500).send('Server Error');
```

Comment:
we usually avoid return 500s straight, since that means you have a straight but unhandled error - what err would occur here? Maybe a 503 as the service is unavailable